zerver/webhooks/github/view.py

```diff
@@ -73,13 +73,18 @@ def get_opened_or_update_pull_request_body(helper: Helper) -> str:
     changes = payload.get("changes", {})
     if "body" in changes or action == "opened":
         description = pull_request["body"].tame(check_none_or(check_string))
+    target_branch = None
+    base_branch = None
+    if action == "opened" or action == "merged":
+        target_branch = pull_request["head"]["label"].tame(check_string)
+        base_branch = pull_request["base"]["label"].tame(check_string)
```

Comment:
Can we do this in the common `git.py` code -- my guess is we have the exact same behavior in the GitLab integration and all the others that use the common `get_pull_request_event_message` function, and so we should probably be addressing the bug there so it works for all of them.

Reply:
I agree that the other integrations should get this same fix if they have the same bug, but I think the bug actually is in the specific integration's code.

The API of this function is — this isn't written down, there's no docstring, but that's a separate issue — that if you pass these two optional parameters, then that "from … to …" text gets included, and if you don't, it doesn't. So the caller is responsible for choosing to include these parameters when it wants that text, and to leave them out when it doesn't.

From a quick search for call sites to this function (`git grep -PA8 'get_pull_request_event_message\('`), generally integrations that use it are making a reasonable attempt at getting that right. Including the GitHub integration — it has some other call sites that don't pass these parameters. So fixing it in other integrations will be a matter of auditing those other call sites to check that they're passing them only when appropriate.

Reply:
Yeah, I agree with Greg. The problem with changing the behaviour in the `get_pull_request_event_message` function is that each integration may have different names for `action`. (E.g Azure Devops uses "created" rather than "opened"). With all these different names for actions, it might overlap and become hard for maintainability. So the best way to go is to edit the integrations rather than editing this common function.

Reply:
Sounds good!